sys-fs/zfs/zfs-2.1.1-r1.ebuild

```diff
@@ -0,0 +1,288 @@
+					eerror "in /usr/src/linux/.config or"
+					eerror "    Device Drivers --->"
+					eerror "        Block devices --->"
+					eerror "            [X] Loopback device support"
+				fi
+			fi
+		fi
+	fi
+}
+
+libsoversion_check() {
+	local bugurl libzfs_sover
+	bugurl="https://bugs.gentoo.org/enter_bug.cgi?form_name=enter_bug&product=Gentoo+Linux&component=Current+packages"
+
+	libzfs_sover="$(grep 'libzfs_la_LDFLAGS += -version-info' lib/libzfs/Makefile.am \
+		| grep -Eo '[0-9]+:[0-9]+:[0-9]+')"
+	libzfs_sover="${libzfs_sover%%:*}"
+
+	if [[ ${libzfs_sover} -ne $(ver_cut 2 ${SLOT}) ]]; then
+		echo
+		eerror "BUG BUG BUG BUG BUG BUG BUG BUG"
+		eerror "ebuild subslot does not match libzfs soversion!"
+		eerror "libzfs soversion: ${libzfs_sover}"
+		eerror "ebuild value: $(ver_cut 2 ${SLOT})"
+		eerror "This is a bug in the ebuild, please use the following URL to report it"
+		eerror "${bugurl}&short_desc=${CATEGORY}%2F${P}+update+subslot"
+		echo
+		# we want to abort for releases, but just print a warning for live ebuild
+		# to keep package installable
+		[[  ${PV} == "9999" ]] || die
+	fi
+}
+
+src_prepare() {
+	default
+	libsoversion_check
+
+	# Run unconditionally (bug #792627)
+	eautoreconf
+
+	if [[ ${PV} != "9999" ]]; then
+		# Set revision number
+		sed -i "s/\(Release:\)\(.*\)1/\1\2${PR}-gentoo/" META || die "Could not set Gentoo release"
+	fi
+
+	if use python; then
+		pushd contrib/pyzfs >/dev/null || die
+		distutils-r1_src_prepare
+		popd >/dev/null || die
+	fi
+
+	# prevent errors showing up on zfs-mount stop, #647688
+	# openrc will unmount all filesystems anyway.
+	sed -i "/^ZFS_UNMOUNT=/ s/yes/no/" "etc/default/zfs.in" || die
+}
+
+src_configure() {
+	use custom-cflags || strip-flags
+	use minimal || python_setup
+
+	local myconf=(
+		--bindir="${EPREFIX}/bin"
+		--enable-shared
+		--enable-systemd
+		--enable-sysvinit
+		--localstatedir="${EPREFIX}/var"
+		--sbindir="${EPREFIX}/sbin"
+		--with-config=user
+		--with-dracutdir="${EPREFIX}/usr/lib/dracut"
+		--with-linux="${KV_DIR}"
+		--with-linux-obj="${KV_OUT_DIR}"
+		--with-udevdir="$(get_udevdir)"
+		--with-pamconfigsdir="${EPREFIX}/unwanted_files"
+		--with-pammoduledir="$(getpam_mod_dir)"
+		--with-systemdunitdir="$(systemd_get_systemunitdir)"
+		--with-systemdpresetdir="${EPREFIX}/lib/systemd/system-preset"
+		--with-vendor=gentoo
+		$(use_enable debug)
+		$(use_enable nls)
+		$(use_enable pam)
+		$(use_enable python pyzfs)
+		--disable-static
+		$(usex minimal --without-python --with-python="${EPYTHON}")
+	)
+
+	econf "${myconf[@]}"
+}
+
+src_compile() {
+	default
+	if use python; then
+		pushd contrib/pyzfs >/dev/null || die
+		distutils-r1_src_compile
+		popd >/dev/null || die
+	fi
+}
+
+src_install() {
+	default
+
+	gen_usr_ldscript -a nvpair uutil zfsbootenv zfs zfs_core zpool
+
+	use pam && { rm -rv "${ED}/unwanted_files" || die ; }
+
+	use test-suite || { rm -r "${ED}/usr/share/zfs/"{test-runner,zfs-tests,runfiles,*sh} || die ; }
```

Comment:
`find "${ED}/usr/share/zfs" -mindepth 1 -maxdepth 1 -not -name compatibility.d -exec rm -r '{}' \;`

this should work better =)

Reply:
Problem is, the PR version makes it easier to not miss things in future.. hm

Reply:
Yeah I considered something like the find, but then when they add more files there we'll drop them again and not notice.
I'm not good with autotools; would it be feasible to instead patch out/disable installing the test files at an earlier stage?

Reply:
```diff
diff --git a/sys-fs/zfs/zfs-2.1.1.ebuild b/sys-fs/zfs/zfs-2.1.1.ebuild
index cf525674e3ab..f141ae27d79e 100644
--- a/sys-fs/zfs/zfs-2.1.1.ebuild
+++ b/sys-fs/zfs/zfs-2.1.1.ebuild
@@ -154,6 +154,13 @@ src_prepare() {
        default
        libsoversion_check
 
+       # remove those before build, and before autoreconf
+       if ! use test-suite; then
+               rm -rf tests || die
+               sed -i '/tests\/.*Makefile/d' configure.ac || die
+               sed -i '/^SUBDIRS\ +=/ s/tests//' Makefile.am || die
+       fi
+
        # Run unconditionally (bug #792627)
        eautoreconf
 
@@ -221,8 +228,6 @@ src_install() {
 
        use pam && { rm -rv "${ED}/unwanted_files" || die ; }
 
-       use test-suite || { rm -r "${ED}/usr/share/zfs" || die ; }
-
        find "${ED}" -name '*.la' -delete || die
 
        dobashcomp contrib/bash_completion.d/zfs
```

this is pre-build deletion version messing with autofiles.
tbh looks fragile, and leaves some extra dangling files too.
I think the PR version is fine in the long run. so let's merge it.

Reply:
I worry about *sh matching something it shouldn't, but at some point, we can try upstream a proper way to not install these. Let's add a comment and merge the PR version?